agents-api/agents_api/activities/tool_executor.py

```diff
@@ -0,0 +1,231 @@
+import json
+from typing import Any, Literal
+
+from pydantic import BaseModel, Field
+
+from ..clients.integrations import run_integration_service
+from ..env import brave_api_key
+
+
+class ToolExecutionResult(BaseModel):
```

Comment:
Also can you please move the models to `agents-api/agents_api/autogen/openapi_model.py`?